crates/router/src/connector/adyen/transformers.rs

```diff
@@ -289,139 +352,341 @@ impl TryFrom<&types::BrowserInformation> for AdyenBrowserInfo {
-            _ => Err(errors::ConnectorError::MissingRequiredField {
-                field_name: "payment_method",
-            }),
-        }?;
-
-        let browser_info = if matches!(item.auth_type, storage_enums::AuthenticationType::ThreeDs) {
-            item.request
-                .browser_info
-                .clone()
-                .map(|d| AdyenBrowserInfo::try_from(&d))
-                .transpose()?
-        } else {
-            None
-        };
+fn get_billing_address(item: &types::PaymentsAuthorizeRouterData) -> Option<Address> {
+    let address = item
+        .address
+        .billing
+        .as_ref()
+        .and_then(|billing| billing.address.as_ref());
+    let city = address.and_then(|address| address.city.clone());
+    let country = address.and_then(|address| address.country.clone());
+    let house_number_or_name = address.and_then(|address| address.line1.clone());
+    let postal_code = address.and_then(|address| address.zip.clone());
+    let state_or_province = address.and_then(|address| address.state.clone());
+    let street = address.and_then(|address| address.line2.clone());
+    Some(Address {
+        city,
+        country,
+        house_number_or_name,
+        postal_code,
+        state_or_province,
+        street,
+    })
+}
 
-        let additional_data = match item.request.capture_method {
-            Some(storage_models::enums::CaptureMethod::Manual) => Some(AdditionalData {
-                authorisation_type: AuthType::PreAuth,
-            }),
-            _ => None,
-        };
+fn get_delivery_address(item: &types::PaymentsAuthorizeRouterData) -> Option<Address> {
+    let address = item
+        .address
+        .shipping
+        .as_ref()
+        .and_then(|shipping| shipping.address.as_ref());
+    let city = address.and_then(|address| address.city.clone());
+    let country = address.and_then(|address| address.country.clone());
+    let house_number_or_name = address.and_then(|address| address.line1.clone());
+    let postal_code = address.and_then(|address| address.zip.clone());
+    let state_or_province = address.and_then(|address| address.state.clone());
+    let street = address.and_then(|address| address.line2.clone());
+    Some(Address {
+        city,
+        country,
+        house_number_or_name,
+        postal_code,
+        state_or_province,
+        street,
+    })
+}
 
-        Ok(Self {
-            amount,
-            merchant_account: auth_type.merchant_account,
-            payment_method,
-            reference,
-            return_url: item.router_return_url.clone().ok_or(
-                errors::ConnectorError::MissingRequiredField {
-                    field_name: "router_return_url",
-                },
-            )?,
-            shopper_interaction,
-            recurring_processing_model,
-            browser_info,
-            additional_data,
+fn get_line_items(item: &types::PaymentsAuthorizeRouterData) -> Option<Vec<LineItem>> {
```

Comment:
Same here, if this function always returns the line item, then do not use option